File: bindings/expo-cpu-scanner/example/modules/frame-camera/android/src/main/java/com/aukilabs/framecamera/FrameCameraModule.kt
package com.aukilabs.framecamera

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class FrameCameraModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("FrameCamera")

    // Preview + analysis surface; emits `onFrame` with Y-plane luma.
    View(FrameCameraView::class) {
      Events("onFrame", "onError")

      Prop("targetFps") { view: FrameCameraView, fps: Double? ->
        view.setTargetFps(fps?.toInt() ?: 12)
      }

      Prop("active") { view: FrameCameraView, active: Boolean? ->
        view.setActive(active ?: true)
      }
    }
  }
}

File: bindings/expo-cpu-scanner/example/modules/frame-camera/ios/FrameCameraModule.swift
import ExpoModulesCore

public class FrameCameraModule: Module {
  public func definition() -> ModuleDefinition {
    Name("FrameCamera")

    View(FrameCameraView.self) {
      Events("onFrame", "onError")

      Prop("targetFps") { (view: FrameCameraView, fps: Double?) in
        view.setTargetFps(Int(fps ?? 12))
      }

      Prop("active") { (view: FrameCameraView, active: Bool?) in
        view.setActive(active ?? true)
      }
    }
  }
}
